policyengine/simulation.py

```diff
@@ -254,44 +284,68 @@ def _initialise_simulations(self):
                 False: USSimulation,
             },
         }[self.country][macro]
-        self.baseline = _simulation_type(
+        self.baseline_sim = _simulation_type(
             dataset=self.data if macro else None,
             situation=self.data if not macro else None,
-            reform=self.baseline,
+            reform=self._parsed_baseline,
         )
-        self.baseline.default_calculation_period = self.time_period
 
-        if "subsample" in self.options:
-            self.baseline = self.baseline.subsample(self.options["subsample"])
+        if "region" in self.options and isinstance(
+            self.baseline_sim, CountryMicrosimulation
+        ):
+            self.baseline_sim = self._apply_region_to_simulation(
+                self.baseline_sim,
+                _simulation_type,
+                self.options["region"],
+                reform=self.baseline_sim.reform,
+            )
+        if self.time_period is not None:
+            self.baseline_sim.default_calculation_period = self.time_period
 
-        if "region" in self.options:
-            self.baseline = self._apply_region_to_simulation(
-                self.baseline, _simulation_type, self.options["region"]
+        if "subsample" in self.options:
+            self.baseline_sim = self.baseline_sim.subsample(
+                self.options["subsample"]
             )
 
         if self.comparison:
-            self.reformed = _simulation_type(
+            if self._parsed_baseline is not None:
+                self._parsed_reform = (self._parsed_baseline, self.reform)
+            self.reformed_sim = _simulation_type(
                 dataset=self.data if macro else None,
                 situation=self.data if not macro else None,
-                reform=self.reform,
+                reform=self._parsed_reform,
             )
-            self.reformed.default_calculation_period = self.time_period
+            if "region" in self.options and isinstance(
+                self.reformed_sim, CountryMicrosimulation
+            ):
+                self.reformed_sim = self._apply_region_to_simulation(
+                    self.reformed_sim,
+                    _simulation_type,
+                    self.options["region"],
+                    reform=self._parsed_reform,
+                )
 
             if "subsample" in self.options:
```

Comment:
__suggestion:__ Modularize code for region selection and subsampling

This code feels a bit duplicative, and I'm wondering if these couldn't be functions, something like `apply_subsample` and `apply_region`. This would also prevent any possible future bug associated with modifying one piece of the code and accidentally forgetting the other. 

This could also be a future thing, don't want to block where not necessary.